### DISCUSSION TITLE
The right using of nested Pinia stores with SSR

### DISCUSSION THREAD

**Discussion by m00nk** [quasarframework/quasar#13490] [CLI - SSR mode] [answered]
I have several pinia stores and some of them use some others. How do I create the stores inside the other stores for good working in SSR? I read I need to send the pinia object (root store object) into the store constructor when I create the stores for SSR like this:

```js
import {createPinia} from "pinia";

import MyStore from 'my-store';

app.use(pinia);

const myStore = MyStore(pinia);
```

But do I need to do the same thing inside the store code to have an ability to use the other stores? Example:

```js
// file my-store.js

import {defineStore} from 'pinia';
import MyOtherStore from 'my-other-store';

export default defineStore('my-store', {
    // ...
    actions: {
         someAction(){
              const other = MyOtherStore(/* 1 */);
         }
    }
}
);
```

Do I need to send the pinia object in (1) for SSR working right? If so, how can I get it?

**Comment by yusufkandemir** [comment, marked as answer]
Answered in https://github.com/vuejs/pinia/discussions/1315#discussioncomment-2799049.
> [posva](https://github.com/posva): No, you don’t need it
